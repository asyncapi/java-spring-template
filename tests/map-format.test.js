const path = require('path');
const Generator = require('@asyncapi/generator');
const { readFile } = require('fs').promises;

const MAIN_TEST_RESULT_PATH = path.join('tests', 'temp', 'integrationTestResult');

const generateFolderName = () => {
    // you always want to generate to new directory to make sure test runs in clear environment
    return path.resolve(MAIN_TEST_RESULT_PATH, Date.now().toString());
};

describe('template integration tests for map format', () => {

    jest.setTimeout(30000);

    it('should generate DTO file with proper map types', async() => {
        const outputDir = generateFolderName();
        const params = {};
        const mapFormatExamplePath = './mocks/map-format.yml';

        const generator = new Generator(path.normalize('./'), outputDir, { forceWrite: true, templateParams: params });
        await generator.generateFromFile(path.resolve('tests', mapFormatExamplePath));

        const expectedFiles = [
            '/src/main/java/com/asyncapi/model/SongMetaData.java',
            '/src/main/java/com/asyncapi/model/SuccessResponse.java',
            '/src/main/java/com/asyncapi/model/FailureResponse.java',
            '/src/main/java/com/asyncapi/model/Interpret.java',
        ];
        for (const index in expectedFiles) {
            const file = await readFile(path.join(outputDir, expectedFiles[index]), 'utf8');
            const fileWithAnyDate = file.replace(/date="\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+([+-][0-2]\d:[0-5]\d|Z)"/, 'date="AnyDate"');
            expect(fileWithAnyDate).toMatchSnapshot();
        }
    });

    it('should generate correct DTO for empty object', async() => {
        const outputDir = generateFolderName();
        const params = {};
        const mapFormatExamplePath = './mocks/map-format.yml';

        const generator = new Generator(path.normalize('./'), outputDir, { forceWrite: true, templateParams: params });
        await generator.generateFromFile(path.resolve('tests', mapFormatExamplePath));

        const expectedFiles = [
            '/src/main/java/com/asyncapi/model/EmptyObject.java',
        ];
        for (const index in expectedFiles) {
            const file = await readFile(path.join(outputDir, expectedFiles[index]), 'utf8');
            const fileWithAnyDate = file.replace(/date="\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+([+-][0-2]\d:[0-5]\d|Z)"/, 'date="AnyDate"');
            expect(fileWithAnyDate).toMatchSnapshot();
        }
    });
});
