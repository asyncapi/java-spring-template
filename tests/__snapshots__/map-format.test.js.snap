// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template integration tests for map format should generate DTO file with proper map types 1`] = `
"package com.asyncapi.model;


import jakarta.validation.constraints.*;
import jakarta.validation.Valid;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonValue;

import javax.annotation.processing.Generated;
import java.util.List;
import java.util.Map;
import java.util.Objects;


@Generated(value="com.asyncapi.generator.template.spring", date="AnyDate")
public class SongMetaData {
    
    private @Valid Map<String, String> tags;
    
    private @Valid Map<String, Long> stats;
    
    private @Valid Map<String, Boolean> flags;
    
    private @Valid Album album;
    
    private @Valid Map<String, Interpret> interprets;
    

    

    /**
     * Tags
     */
    @JsonProperty("tags")
    public Map<String, String> getTags() {
        return tags;
    }

    public void setTags(Map<String, String> tags) {
        this.tags = tags;
    }
    

    /**
     * Stats
     */
    @JsonProperty("stats")
    public Map<String, Long> getStats() {
        return stats;
    }

    public void setStats(Map<String, Long> stats) {
        this.stats = stats;
    }
    

    /**
     * Flags
     */
    @JsonProperty("flags")
    public Map<String, Boolean> getFlags() {
        return flags;
    }

    public void setFlags(Map<String, Boolean> flags) {
        this.flags = flags;
    }
    

    /**
     * Album
     */
    @JsonProperty("album")
    public Album getAlbum() {
        return album;
    }

    public void setAlbum(Album album) {
        this.album = album;
    }
    

    /**
     * Interprets
     */
    @JsonProperty("interprets")
    public Map<String, Interpret> getInterprets() {
        return interprets;
    }

    public void setInterprets(Map<String, Interpret> interprets) {
        this.interprets = interprets;
    }
    
    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        SongMetaData songMetaData = (SongMetaData) o;
        return 
            Objects.equals(this.tags, songMetaData.tags) &&
            Objects.equals(this.stats, songMetaData.stats) &&
            Objects.equals(this.flags, songMetaData.flags) &&
            Objects.equals(this.album, songMetaData.album) &&
            Objects.equals(this.interprets, songMetaData.interprets);
    }

    @Override
    public int hashCode() {
        return Objects.hash(tags, stats, flags, album, interprets);
    }

    @Override
    public String toString() {
        return "class SongMetaData {\\n" +
        
                "    tags: " + toIndentedString(tags) + "\\n" +
                "    stats: " + toIndentedString(stats) + "\\n" +
                "    flags: " + toIndentedString(flags) + "\\n" +
                "    album: " + toIndentedString(album) + "\\n" +
                "    interprets: " + toIndentedString(interprets) + "\\n" +
                "}";
    }

    /**
     * Convert the given object to string with each line indented by 4 spaces (except the first line).
     */
    private String toIndentedString(Object o) {
        if (o == null) {
           return "null";
        }
        return o.toString().replace("\\n", "\\n    ");
    }
}"
`;

exports[`template integration tests for map format should generate DTO file with proper map types 2`] = `
"package com.asyncapi.model;


import jakarta.validation.constraints.*;
import jakarta.validation.Valid;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonValue;

import javax.annotation.processing.Generated;
import java.util.List;
import java.util.Map;
import java.util.Objects;


@Generated(value="com.asyncapi.generator.template.spring", date="AnyDate")
public class SuccessResponse {
    
    private @Valid String originalEventId;
    
    private @Valid Boolean success;
    
    private @Valid Map<String, Object> meta;
    

    

    /**
     * Id of the original Event
     */
    @JsonProperty("originalEventId")
    public String getOriginalEventId() {
        return originalEventId;
    }

    public void setOriginalEventId(String originalEventId) {
        this.originalEventId = originalEventId;
    }
    

    /**
     * Shows whether or not the original Event was processed correctly
     */
    @JsonProperty("success")
    public Boolean getSuccess() {
        return success;
    }

    public void setSuccess(Boolean success) {
        this.success = success;
    }
    

    /**
     * Meta-Information
     */
    @JsonProperty("meta")
    public Map<String, Object> getMeta() {
        return meta;
    }

    public void setMeta(Map<String, Object> meta) {
        this.meta = meta;
    }
    
    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        SuccessResponse successResponse = (SuccessResponse) o;
        return 
            Objects.equals(this.originalEventId, successResponse.originalEventId) &&
            Objects.equals(this.success, successResponse.success) &&
            Objects.equals(this.meta, successResponse.meta);
    }

    @Override
    public int hashCode() {
        return Objects.hash(originalEventId, success, meta);
    }

    @Override
    public String toString() {
        return "class SuccessResponse {\\n" +
        
                "    originalEventId: " + toIndentedString(originalEventId) + "\\n" +
                "    success: " + toIndentedString(success) + "\\n" +
                "    meta: " + toIndentedString(meta) + "\\n" +
                "}";
    }

    /**
     * Convert the given object to string with each line indented by 4 spaces (except the first line).
     */
    private String toIndentedString(Object o) {
        if (o == null) {
           return "null";
        }
        return o.toString().replace("\\n", "\\n    ");
    }
}"
`;

exports[`template integration tests for map format should generate DTO file with proper map types 3`] = `
"package com.asyncapi.model;


import jakarta.validation.constraints.*;
import jakarta.validation.Valid;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonValue;

import javax.annotation.processing.Generated;
import java.util.List;
import java.util.Map;
import java.util.Objects;


@Generated(value="com.asyncapi.generator.template.spring", date="AnyDate")
public class FailureResponse {
    
    private @Valid String originalEventId;
    
    private @Valid Boolean success;
    
    private @Valid Map<String, String> meta;
    

    

    /**
     * Id of the original Event
     */
    @JsonProperty("originalEventId")
    public String getOriginalEventId() {
        return originalEventId;
    }

    public void setOriginalEventId(String originalEventId) {
        this.originalEventId = originalEventId;
    }
    

    /**
     * Shows whether or not the original Event was processed correctly
     */
    @JsonProperty("success")
    public Boolean getSuccess() {
        return success;
    }

    public void setSuccess(Boolean success) {
        this.success = success;
    }
    

    /**
     * Meta-Information
     */
    @JsonProperty("meta")
    public Map<String, String> getMeta() {
        return meta;
    }

    public void setMeta(Map<String, String> meta) {
        this.meta = meta;
    }
    
    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        FailureResponse failureResponse = (FailureResponse) o;
        return 
            Objects.equals(this.originalEventId, failureResponse.originalEventId) &&
            Objects.equals(this.success, failureResponse.success) &&
            Objects.equals(this.meta, failureResponse.meta);
    }

    @Override
    public int hashCode() {
        return Objects.hash(originalEventId, success, meta);
    }

    @Override
    public String toString() {
        return "class FailureResponse {\\n" +
        
                "    originalEventId: " + toIndentedString(originalEventId) + "\\n" +
                "    success: " + toIndentedString(success) + "\\n" +
                "    meta: " + toIndentedString(meta) + "\\n" +
                "}";
    }

    /**
     * Convert the given object to string with each line indented by 4 spaces (except the first line).
     */
    private String toIndentedString(Object o) {
        if (o == null) {
           return "null";
        }
        return o.toString().replace("\\n", "\\n    ");
    }
}"
`;

exports[`template integration tests for map format should generate DTO file with proper map types 4`] = `
"package com.asyncapi.model;


import jakarta.validation.constraints.*;
import jakarta.validation.Valid;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonValue;

import javax.annotation.processing.Generated;
import java.util.List;
import java.util.Map;
import java.util.Objects;


@Generated(value="com.asyncapi.generator.template.spring", date="AnyDate")
public class Interpret {
    
    private @Valid String name;
    
    private @Valid Map<String, Object> meta;
    

    

    /**
     * Interpret name
     */
    @JsonProperty("name")
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
    

    /**
     * Meta-Information
     */
    @JsonProperty("meta")
    public Map<String, Object> getMeta() {
        return meta;
    }

    public void setMeta(Map<String, Object> meta) {
        this.meta = meta;
    }
    
    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        Interpret interpret = (Interpret) o;
        return 
            Objects.equals(this.name, interpret.name) &&
            Objects.equals(this.meta, interpret.meta);
    }

    @Override
    public int hashCode() {
        return Objects.hash(name, meta);
    }

    @Override
    public String toString() {
        return "class Interpret {\\n" +
        
                "    name: " + toIndentedString(name) + "\\n" +
                "    meta: " + toIndentedString(meta) + "\\n" +
                "}";
    }

    /**
     * Convert the given object to string with each line indented by 4 spaces (except the first line).
     */
    private String toIndentedString(Object o) {
        if (o == null) {
           return "null";
        }
        return o.toString().replace("\\n", "\\n    ");
    }
}"
`;

exports[`template integration tests for map format should generate correct DTO for empty object 1`] = `
"package com.asyncapi.model;


import jakarta.validation.constraints.*;
import jakarta.validation.Valid;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonValue;

import javax.annotation.processing.Generated;
import java.util.List;
import java.util.Map;
import java.util.Objects;

/**
 * Test correct generation of object without parameters
 */
@Generated(value="com.asyncapi.generator.template.spring", date="AnyDate")
public class EmptyObject {
    

    
    @Override
    public boolean equals(Object o) {
        return super.equals(o);
    }

    @Override
    public int hashCode() {
        return super.hashCode();
    }

    @Override
    public String toString() {
        return "class EmptyObject {\\n" +
        
                "}";
    }

    /**
     * Convert the given object to string with each line indented by 4 spaces (except the first line).
     */
    private String toIndentedString(Object o) {
        if (o == null) {
           return "null";
        }
        return o.toString().replace("\\n", "\\n    ");
    }
}"
`;
